core/crypto/sm4.c

```diff
@@ -0,0 +1,263 @@
+// SPDX-License-Identifier: BSD-2-Clause
+/*
+ * Copyright Copyright (c) 2019 Huawei Technologies Co., Ltd
+ */
+/*
+ * SM4 Encryption algorithm (SMS4 algorithm)
+ * GM/T 0002-2012 Chinese National Standard ref:http://www.oscca.gov.cn/
+ * thanks to Xyssl
+ * thnaks and refers to http://hi.baidu.com/numax/blog/item/80addfefddfb93e4cf1b3e61.html
+ * author:goldboar
+ * email:goldboar@163.com
+ * 2012-4-20
+ */
+
+#include "sm4.h"
+#include <string.h>
+
+#define GET_UINT32_BE(n, b, i)			\
```

Comment:
We have `get_be32()` in `<io.h>` for this.

Reply:
Right, I will use it.